spec/lib/blackout_updater_spec.rb

```diff
@@ -0,0 +1,66 @@
+        blackout = instance_spy(Blackout, update_attributes: nil)
+        allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                    :active).and_return([])
+        updater = BlackoutUpdater.new(blackout: blackout, params: {})
+        updater.update!
+        expect(blackout).to have_received(:update_attributes)
+      end
+      it 'returns success message' do
+        blackout = instance_spy(Blackout, update_attributes: nil)
+        allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                    :active).and_return([])
+        updater = BlackoutUpdater.new(blackout: blackout, params: {})
+        results = updater.update!
+        expect(results[:result]).to eq('Blackout was successfully updated.')
+      end
+      it 'does not return an error' do
+        blackout = instance_spy(Blackout, update_attributes: nil)
+        allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                    :active).and_return([])
+        updater = BlackoutUpdater.new(blackout: blackout, params: {})
+        results = updater.update!
+        expect(results[:error]).to be_nil
+      end
+    end
+    context 'with errors' do
+      context 'with conflicting blackout' do
+        it 'does not update blackout' do
+          blackout = instance_spy(Blackout, update_attributes: nil)
+          cf = double('conflict', md_link: 'Test')
+          allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                      :active).and_return([cf])
+          updater = BlackoutUpdater.new(blackout: blackout, params: {})
+          updater.update!
+          expect(blackout).not_to have_received(:update_attributes)
+        end
+        it 'does not return a result' do
+          blackout = instance_spy(Blackout, update_attributes: nil)
+          cf = double('conflict', md_link: 'Test')
+          allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                      :active).and_return([cf])
+          updater = BlackoutUpdater.new(blackout: blackout, params: {})
+          results = updater.update!
+          expect(results[:result]).to be_nil
+        end
+        it 'returns an error' do
+          blackout = instance_spy(Blackout, update_attributes: nil)
+          cf = double('conflict', md_link: 'Test')
+          allow(Reservation).to receive_message_chain(:overlaps_with_date_range,
+                                                      :active).and_return([cf])
+          updater = BlackoutUpdater.new(blackout: blackout, params: {})
+          results = updater.update!
+          expect(results[:error]).to eq('The following reservation(s) will be'\
+            ' unable to be returned: Test. Please update their due dates and '\
+            'try again.')
```

Comment:
instead of testing the whole string, test if it includes the reservation's md_link